src/plugins/intel_npu/tests/functional/behavior/npuw/use_case_tests.cpp

```diff
@@ -0,0 +1,261 @@
+    EXPECT_ANY_THROW(compiled_model.create_infer_request());
+}
+
+TEST_F(NPUWUseCaseTests, InferIsSuccessful) {
+    // Set expectations first:
+    EXPECT_CALL(*npu_plugin, compile_model(A<const std::shared_ptr<const ov::Model>&>(), _)).Times(1);
+    EXPECT_CALL(*cpu_plugin, compile_model(A<const std::shared_ptr<const ov::Model>&>(), _)).Times(0);
+    npu_plugin->set_expectations_to_comp_models(0, [](MockCompiledModel& model) {
+        EXPECT_CALL(model, create_sync_infer_request()).Times(1);
+    });
+    npu_plugin->set_expectations_to_infer_reqs(0, [](MockInferRequest& request) {
+        EXPECT_CALL(request, infer()).Times(1);
+    });
+
+    // Register mock objects as plugins in OpenVINO:
+    register_mock_plugins_in_ov();
+
+    // Do the actual test:
+    auto npuw_properties = 
+        ov::AnyMap{ov::intel_npu::use_npuw(true),
+                   ov::intel_npu::npuw::devices("MockNPU")};
+    auto compiled_model = core.compile_model(model, "NPU", npuw_properties);
+    auto infer_request = compiled_model.create_infer_request();
+    EXPECT_NO_THROW(infer_request.infer());
+}
+
+TEST_F(NPUWUseCaseTests, InferIsFailSafe) {
+    // Set expectations first:
+    {
+        InSequence seq;
+        EXPECT_CALL(*npu_plugin, compile_model(A<const std::shared_ptr<const ov::Model>&>(), _)).Times(1);
+        EXPECT_CALL(*cpu_plugin, compile_model(A<const std::shared_ptr<const ov::Model>&>(), _)).Times(1);
+    }
+    npu_plugin->set_expectations_to_comp_models(0, [](MockCompiledModel& model) {
+        EXPECT_CALL(model, create_sync_infer_request()).Times(1);
+    });
+    npu_plugin->set_expectations_to_infer_reqs(0, [](MockInferRequest& request) {
+        EXPECT_CALL(request, infer())
+            .Times(1)
+            .WillOnce(Throw(std::runtime_error("Infer on MockNPU is failed")));
+    });   
+    cpu_plugin->set_expectations_to_comp_models(0, [](MockCompiledModel& model) {
+        EXPECT_CALL(model, create_sync_infer_request()).Times(1);
+    });
+    cpu_plugin->set_expectations_to_infer_reqs(0, [](MockInferRequest& request) {
+        EXPECT_CALL(request, infer()).Times(1);
+    });
+
+    // Register mock objects as plugins in OpenVINO:
+    register_mock_plugins_in_ov();
+
+    // Do the actual test:
+    auto npuw_properties = 
+        ov::AnyMap{ov::intel_npu::use_npuw(true),
+                   ov::intel_npu::npuw::devices("MockNPU,MockCPU")};
+    auto compiled_model = core.compile_model(model, "NPU", npuw_properties);
+    auto infer_request = compiled_model.create_infer_request();
+    EXPECT_NO_THROW(infer_request.infer());
+}
+
+TEST_F(NPUWUseCaseTests, InferIsFailed) {
+    // Set expectations first:
+    {
+        InSequence seq;
+        EXPECT_CALL(*npu_plugin, compile_model(A<const std::shared_ptr<const ov::Model>&>(), _)).Times(1);
+        EXPECT_CALL(*cpu_plugin, compile_model(A<const std::shared_ptr<const ov::Model>&>(), _)).Times(0);
```

Comment:
@AsyaPronina could you please explain me what's the difference in those `infer` tests above and how they fail or pass. The only major difference I see is `Times(0)`/`Times(1)` in `cpu_plugin`. What plugins/devices will be available for those tests? Do those tests run in precommit?

Reply:
These tests run on `MockCpu` and `MockCpu` devices and plugins. The major difference in setting of expectations for procedure. For example for `InferIsFailed` test I've set `MockNpu` plugin `infer_request.infer()` call to fail:

```
    npu_plugin->set_expectations_to_infer_reqs(0, [](MockInferRequest& request) {
        EXPECT_CALL(request, infer())
            .Times(1)
            .WillOnce(Throw(std::runtime_error("Infer on MockNPU is failed")));
    });
```

And you can see, that `EXPECT_CALL` for `cpu_plugin.compile_model()` is set to `0` here. That means NPUW will  not failover to CPU failure of inference on NPU. It is what we expect, because I've set only `MockNPU` as NPUW devices here with:

```
auto npuw_properties = 
        ov::AnyMap{ov::intel_npu::use_npuw(true),
                   ov::intel_npu::npuw::devices("MockNPU")};
```

Reply:
Will those run in precommit in this PR or is it a separate activity? If it's separate, feel free to resolve this comment. Otherwise, I suggest we keep it

Reply:
It will run in precommit, let us wait for CI to finish

Reply:
This results from Windows testing:

```
[==========] Running 1418 tests from 158 test suites.
[----------] Global test environment set-up.
[ PG INFO  ] PostgreSQL Reporting is disabled due to missing environment settings
[----------] 10 tests from BehaviorTestsNPUW
[ RUN      ] BehaviorTestsNPUW.TestInfrastructureIsCorrect
[       OK ] BehaviorTestsNPUW.TestInfrastructureIsCorrect (53 ms)
[ RUN      ] BehaviorTestsNPUW.CompilationIsSuccessful
[       OK ] BehaviorTestsNPUW.CompilationIsSuccessful (25 ms)
[ RUN      ] BehaviorTestsNPUW.CompilationIsFailSafe
[       OK ] BehaviorTestsNPUW.CompilationIsFailSafe (23 ms)
[ RUN      ] BehaviorTestsNPUW.CompilationIsFailed
[       OK ] BehaviorTestsNPUW.CompilationIsFailed (21 ms)
[ RUN      ] BehaviorTestsNPUW.InferRequestCreationIsSuccessful
[       OK ] BehaviorTestsNPUW.InferRequestCreationIsSuccessful (21 ms)
[ RUN      ] BehaviorTestsNPUW.InferRequestCreationIsFailSafe
[       OK ] BehaviorTestsNPUW.InferRequestCreationIsFailSafe (22 ms)
[ RUN      ] BehaviorTestsNPUW.InferRequestCreationIsFailed
[       OK ] BehaviorTestsNPUW.InferRequestCreationIsFailed (24 ms)
[ RUN      ] BehaviorTestsNPUW.InferIsSuccessful
[       OK ] BehaviorTestsNPUW.InferIsSuccessful (24 ms)
[ RUN      ] BehaviorTestsNPUW.InferIsFailSafe
[       OK ] BehaviorTestsNPUW.InferIsFailSafe (22 ms)
[ RUN      ] BehaviorTestsNPUW.InferIsFailed
[       OK ] BehaviorTestsNPUW.InferIsFailed (22 ms)
[----------] 10 tests from BehaviorTestsNPUW (258 ms total)
```


These results are from Linux testing:
```
[==========] Running 1421 tests from 159 test suites.
[----------] Global test environment set-up.
[ PG INFO  ] PostgreSQL Reporting is disabled due to missing environment settings
[----------] 10 tests from BehaviorTestsNPUW
[ RUN      ] BehaviorTestsNPUW.TestInfrastructureIsCorrect
[       OK ] BehaviorTestsNPUW.TestInfrastructureIsCorrect (4 ms)
[ RUN      ] BehaviorTestsNPUW.CompilationIsSuccessful
[       OK ] BehaviorTestsNPUW.CompilationIsSuccessful (4 ms)
[ RUN      ] BehaviorTestsNPUW.CompilationIsFailSafe
[       OK ] BehaviorTestsNPUW.CompilationIsFailSafe (5 ms)
[ RUN      ] BehaviorTestsNPUW.CompilationIsFailed
[       OK ] BehaviorTestsNPUW.CompilationIsFailed (3 ms)
[ RUN      ] BehaviorTestsNPUW.InferRequestCreationIsSuccessful
[       OK ] BehaviorTestsNPUW.InferRequestCreationIsSuccessful (3 ms)
[ RUN      ] BehaviorTestsNPUW.InferRequestCreationIsFailSafe
[       OK ] BehaviorTestsNPUW.InferRequestCreationIsFailSafe (5 ms)
[ RUN      ] BehaviorTestsNPUW.InferRequestCreationIsFailed
[       OK ] BehaviorTestsNPUW.InferRequestCreationIsFailed (3 ms)
[ RUN      ] BehaviorTestsNPUW.InferIsSuccessful
[       OK ] BehaviorTestsNPUW.InferIsSuccessful (3 ms)
[ RUN      ] BehaviorTestsNPUW.InferIsFailSafe
[       OK ] BehaviorTestsNPUW.InferIsFailSafe (5 ms)
[ RUN      ] BehaviorTestsNPUW.InferIsFailed
[       OK ] BehaviorTestsNPUW.InferIsFailed (3 ms)
[----------] 10 tests from BehaviorTestsNPUW (38 ms total)
```